Additional MONs and MGRs are no longer deployed by `ceph-salt`

ricardoasmarques opened the pull request:
`ceph orch apply -i <service_spec.yaml>` should be used instead

Signed-off-by: Ricardo Marques <rimarques@suse.com>

To Do:

- [ ] Improve the config tree (all bootstrap configs must be moved to `/Bootstrap/..`)

Commit:
Additional MONs and MGRs are no longer deployed by `ceph-salt`

`ceph orch apply -i <service_spec.yaml>` should be used instead

Signed-off-by: Ricardo Marques <rimarques@suse.com>

smithfarm commented:
Note: https://github.com/SUSE/sesdev/pull/189 is needed to test this with `sesdev`.

Commit:
Improve ceph-salt config tree structure

- /Cluster renamed to /Ceph_Cluster
- Bootstrap related configs moved to `/Cephadm_Bootstrap`
- Created "Bootstrap" role

Signed-off-by: Ricardo Marques <rimarques@suse.com>

ricardoasmarques commented:
Pushed a new commit with my proposal to move all bootstrap related configs to a dedicated tree node:

```
master:~ # ceph-salt config ls
o- / ................................................................................................... [...]
  o- Cluster ........................................................................................... [...]
  | o- Minions .................................................................................. [Minions: 1]
  | | o- master.ses7-single-tmp.com .................................................................. [admin]
  | o- Roles ........................................................................................... [...]
  |   o- Admin .................................................................................. [Minions: 1]
  |     o- master.ses7-single-tmp.com ....................................................... [no other roles]
  o- Cluster_Bootstrap ............................................................................. [enabled]
  | o- Ceph_Conf ....................................................................................... [...]
  | | o- global ........................................................................................ [...]
  | |   o- osd crush chooseleaf type ..................................................................... [0]
  | o- Dashboard ....................................................................................... [...]
  | | o- password ...................................................................................... [***]
  | | o- username .................................................................................... [admin]
  | o- Minion ................................................................... [master.ses7-single-tmp.com]
  o- Containers ........................................................................................ [...]
  | o- Images .......................................................................................... [...]
  |   o- ceph ................................ [registry.suse.de/devel/storage/7.0/containers/ses/7/ceph/ceph]
  o- SSH ...................................................................................... [Key Pair set]
  | o- Private_Key ......................................... [32:0e:d0:36:e7:4a:fc:c0:fe:29:9d:7c:bb:fb:d3:e9]
  | o- Public_Key .......................................... [32:0e:d0:36:e7:4a:fc:c0:fe:29:9d:7c:bb:fb:d3:e9]
  o- System_Update ..................................................................................... [...]
  | o- Packages .................................................................................... [enabled]
  | o- Reboot ..................................................................................... [disabled]
  o- Time_Server .................................................................................. [disabled]
    o- External_Servers .................................................................................. [1]
    | o- 0.pt.pool.ntp.org ............................................................................. [...]
    o- Server_Hostname .......................................................... [master.ses7-single-tmp.com]
```

smithfarm commented:
Would it be worth considering to fold `Cluster_Bootstrap` into `Cluster`?

i.e.

```
o- /
  o- Cluster
    o- Bootstrap
```

instead of the current

```
o- /
  o- Cluster
  o- Cluster_Bootstrap 
```

smithfarm commented:
Another idea, just throwing it out there:

Could "Bootstrap" become a Role, limited to a single minion (like "Admin" used to be)?

smithfarm commented on ceph_salt/config_shell.py:
```diff
@@ -338,32 +359,22 @@ def possible_values(self):
         }
     },
-    'Deployment': {
+    'Cluster_Bootstrap': {
```
maybe `Cephadm_Bootstrap` would be more explicit and easier for the user to understand?

and `Ceph_Cluster` instead of `Cluster`

Just an idea.

ricardoasmarques commented:
@smithfarm I've pushed a new proposal:

```
o- / ......................................................................................................................... [...]
  o- Ceph_Cluster ............................................................................................................ [...]
  | o- Minions ........................................................................................................ [Minions: 1]
  | | o- master.ses7-single-tmp.com ............................................................................. [bootstrap, admin]
  | o- Roles ................................................................................................................. [...]
  |   o- Admin ........................................................................................................ [Minions: 1]
  |   | o- master.ses7-single-tmp.com ..................................................................... [other roles: bootstrap]
  |   o- Bootstrap .................................................................................... [master.ses7-single-tmp.com]
  o- Cephadm_Bootstrap ................................................................................................... [enabled]
  | o- Ceph_Conf ............................................................................................................. [...]
  | o- Dashboard ............................................................................................................. [...]
  |   o- password ............................................................................................. [randomly generated]
  |   o- username .......................................................................................................... [admin]
  o- Containers .............................................................................................................. [...]
  | o- Images ................................................................................................................ [...]
  |   o- ceph ...................................................... [registry.suse.de/devel/storage/7.0/containers/ses/7/ceph/ceph]
  o- SSH ............................................................................................................ [Key Pair set]
  | o- Private_Key ............................................................... [60:9f:3f:76:83:9f:b1:7b:aa:51:54:60:cc:1f:f6:77]
  | o- Public_Key ................................................................ [60:9f:3f:76:83:9f:b1:7b:aa:51:54:60:cc:1f:f6:77]
  o- System_Update ........................................................................................................... [...]
  | o- Packages ......................................................................................................... [disabled]
  | o- Reboot ........................................................................................................... [disabled]
  o- Time_Server ......................................................................................................... [enabled]
    o- External_Servers ........................................................................................................ [1]
    | o- 0.pt.pool.ntp.org ................................................................................................... [...]
    o- Server_Hostname ................................................................................ [master.ses7-single-tmp.com]
```

smithfarm reviewed (approved):
Nice!

smithfarm commented:
Tested with https://github.com/SUSE/sesdev/pull/189 using the following command:

```
sesdev create octopus --ceph-salt-repo https://github.com/ricardoasmarques/ceph-salt --ceph-salt-branch drop-mon-mgr-deployment --single-node --qa-test octopus
```

Test result: PASS
